src/Previews.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import AppBar from '@material-ui/core/AppBar';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Typography from '@material-ui/core/Typography';
+import Box from '@material-ui/core/Box';
+import { getPreviews } from './queueserver'
+
+interface TabPanelProps {
+  children?: React.ReactNode;
+  index: any;
+  value: any;
+}
+
+function TabPanel(props: TabPanelProps) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`scrollable-auto-tabpanel-${index}`}
+      aria-labelledby={`scrollable-auto-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Box p={3}>
+          <Typography>{children}</Typography>
+        </Box>
+      )}
+    </div>
+  );
+}
+
+type PreviewsProps = {
+  runUid: string;
+}
+  
+type PreviewsState = {
+  value: number;
+  previews: string[];
+  intervalId: any;
+}
+
+export class Previews extends React.Component<PreviewsProps, PreviewsState> {
+
+  public constructor(props: PreviewsProps) {
+    super(props);
+    this.state = {
+      value: 0,
+      previews: [],
+      intervalId: 0
+    };
+  }
+
+  private handleChange = (event: React.ChangeEvent<{}>, newValue: number) => {
+    this.setState({value: newValue});
+  };
+
+  private getPreviewsInternal(){
+      if (this.props.runUid){
+        getPreviews(this.props.runUid).then((result) => {
+          this.setState({previews: result})
+        })
+      }
+  }
+
+  componentDidMount(){
+    this.getPreviewsInternal.bind(this)
+    this.setState({intervalId: setInterval(this.getPreviewsInternal.bind(this), 500)});
```

Comment:
Maybe it makes sense to poll less aggressively. Ideally, it should be a configurable option. But for now, let's have 1 second for the interval of polling
```suggestion
    this.setState({intervalId: setInterval(this.getPreviewsInternal.bind(this), 1000)});
```